### PR TITLE
Test plan

### PR DESCRIPTION
**Pull request by MaximeTAlgosup**

### CODE REVIEW

**Review by benoitdekeyn-algosup — APPROVED** [review]

File: documents/QA/testing/random_generator.asm [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+set rd 12345
+
+random:
+// future result in re
+addi re ra 0
+addi rf rb 0
+addi rg rc 0
+mul re rg
+add re rd
+div re rf           // TODO MODULO
```
Comment (leo-chartier) [inline documents/QA/testing/random_generator.asm:16]: Don't forget to do the modulo

File: documents/QA/test_cases.md
Comment (leo-chartier) [inline documents/QA/test_cases.md:1]: The file is passed at the same time you run the program. As such, test case 1 would either return an error or a usage message.

Also, specify if you are running the assembler or emulator.

File: documents/QA/test_plan.md [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+
+#### In-scope
+
+- Assembly instruction:
+  - All the instructions defined in the [Functional Specification](/documents/functional/functional_specification.md) should function as intended.
+- Assembler:
+  - Absence of significant bug;
+  - Should run properly;
+  - Should compile the files;
+  - Should interpret ASM files;
```
Comment (leo-chartier) [inline documents/QA/test_plan.md:123]: You should split the scope of the assembler and emulator.

File: documents/QA/test_plan.md [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+  - Windows 11 Pro
+  - 14-inch (1920 x 1080)
+  - Processor: 13th Gen Intel(R) Core(TM) i7-1355U, 1700 Mhz
+  - RAM: 16.0 GB
+  - System type: 64-bit operating system, x64-based processor
+
+### Software
+
+- [Visual Studio Code](https://code.visualstudio.com/) for the development of the product.
+- [GitHub](https://github.com) for the version control and to track the defects.
```
Comment (leo-chartier) [inline documents/QA/test_plan.md:196]: The actual version control is not GitHub but git.

**Review by leo-chartier — CHANGES_REQUESTED** [review]

**Pull request merged**